development/code-guidelines/tasks-metatasks-guidelines.md

```diff
@@ -0,0 +1,138 @@
+# Tasks and MetaTasks Coding Guidelines
+
+## Introduction
+
+OSC Takss and metatask classes are used to synchronize information across different systems. This document outlines the code conventions and guidelines that must be followed when authoring tasks and metatasks. 
+
+## Conventions
+
+### Location
+
+All tasks and metatasks classes must be in a package named with the prefix **org.osc.core.broker.service.tasks.***.
+> Note:  Only common classes accross many tasks or metatasks should be placed directly in **org.osc.core.broker.service.tasks**, most classes should be under a more specific package.  
+
+**DO:**
+
+```java
+org.osc.core.broker.service.tasks.securitymanager.CreateSecurityManagerDeviceMemberTask 
+```
+
+**DON'T:**
+ 
+```java
+org.osc.core.broker.service.securitymanager.CreateSecurityManagerDeviceMemberTask
+```
+```java
+org.osc.core.broker.service.tasks.CreateSecurityManagerDeviceMemberTask
+```
+
+### Naming
+
+#### Package Names
+As previously mentioned all tasks and metataks packages must be prefixed with **org.osc.core.broker.service.tasks***. In addition to that they must follow the scheme: `org.osc.core.broker.service.tasks.[OPT:VIRTUALIZATION_PLATFORM].[OPT:INTEGRATION_POINT].[ENTITY_NAME]`.  
+* `VIRTUALIZATION_PLATFORM`: The possible values for this *optional* part are `ost` and `k8s`. This part indicates that the task or metatask in the package is specific to OpenStack (`ost`) or Kubernete (`k8s`).  Only tasks and metatasks agnostic to the virtualization platform should not have this part.  
+*  `INTEGRATION_POINT`: The possible values for this *optional* part are `securitymanager` and `sdncontroller`. This part indicates that the task or metetask in the package is specific to security managers or SDN controllers.  Only tasks and metatasks agnostic to an specific integration point should not have this part.  
```

Comment:
to make the package names less verbose, can we use "sdn" for sdncontroller and "manager" for securitymanager

Reply:
IMO those are a bit too generic, specially `manager`.  `manager` is a term commonly used in code bases for various things. True fact: when I did the initial ramp up I thought `manager` was where we kept our manager like classes.  
`sdn` is OK but  since the integration point is really a controller I think `sdncontroller` is more explicit also it  is similar to  `securiymanager` in both length and meaning.